aram/src/main/java/fr/bmartel/aram/RuleEntry.java

```diff
@@ -170,6 +170,21 @@ private void recycle() {
         deleted = this;
     }
 
+    /**
+     * delete all rules
+     *
+     */
+    static void deleteAll() {
+        JCSystem.beginTransaction();
```

Comment:
When handling transactions, I would keep the scope as small and tiny as possible, only including actual write operations that would guarantee functioning. For a chain like this, the shortest atomic operation should be removal of a single item.

The whole rule management looks a bit sketchy but would probably require further reading of the spec to understand usage.